playground/backend/internal/preparators/java_preparators.go

```diff
@@ -37,15 +39,55 @@ const (
 
 // GetJavaPreparators returns preparation methods that should be applied to Java code
 func GetJavaPreparators(filePath string) *[]Preparator {
-	publicClassModification := Preparator{
-		Prepare: replace,
+	removePublicClassPreparator := Preparator{
+		Prepare: removePublicClass,
 		Args:    []interface{}{filePath, classWithPublicModifierPattern, classWithoutPublicModifierPattern},
 	}
-	additionalPackage := Preparator{
-		Prepare: replace,
+	changePackagePreparator := Preparator{
+		Prepare: changePackage,
 		Args:    []interface{}{filePath, packagePattern, importStringPattern},
 	}
-	return &[]Preparator{publicClassModification, additionalPackage}
+	removePackagePreparator := Preparator{
+		Prepare: removePackage,
+		Args:    []interface{}{filePath, packagePattern, newLinePattern},
+	}
+	return &[]Preparator{removePublicClassPreparator, changePackagePreparator, removePackagePreparator}
+}
+
+//removePublicClass changes the 'public class' in the code to 'class'
+func removePublicClass(args ...interface{}) error {
+	err := replace(args...)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//changePackage changes the 'package' to 'import' and the last directory in the package value to '*'
+func changePackage(args ...interface{}) error {
+	valRes := args[3].(*sync.Map)
+	isKata, ok := valRes.Load(validators.KatasValidatorName)
+	if ok && isKata.(bool) {
+		return nil
+	}
+	err := replace(args...)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
I'm just wondering, is this the same as the code below?
```go
err := replace(args...)
return err
```

Reply:
Yes, this functions are wrappers on _replace_ method

Reply:
Done